libs/dal/src/repositories/execution-details/execution-details.schema.ts

```diff
@@ -79,6 +70,12 @@ const executionDetailsSchema = new Schema<ExecutionDetailsDBModel>(
   schemaOptions
 );
 
+executionDetailsSchema.index({
+  _messageId: 1,
+  _jobId: 1,
+  _environmentId: 1,
+});
+
```

Comment:
Would love to raise this index because he is very very heavy index storage-wise( every index is more than 1G ), therefore I tried to minimize the amount of the indexes. In addition, IMO most of the fetches are on notification.populate which is under _id.

I left indexes {  _messageId: 1,  _jobId: 1} because of the use of executionDetailsRepository.count in MessageMatcher.

The addition indexes that could have need added are:
  webhookStatus: 1,
  _notificationId: 1,
  _notificationTemplateId: 1,
  _subscriberId: 1,
  transactionId: 1,